cmd/thriftbreak/main.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"go.uber.org/multierr"
+	"go.uber.org/thriftrw/compile"
+	"go.uber.org/thriftrw/internal/compare"
+)
+
+func main() {
+	if err := run(os.Args[1:]); err != nil {
+		log.Fatalf("linter error: %v", err)
+	}
+}
+
+func run(args []string) error {
+
+	fs := flag.NewFlagSet("thriftcompat", flag.ContinueOnError)
+	toFile := fs.String("to_file", "", "updated file")
+	fromFile := fs.String("from_file", "", "original file")
+	if err := fs.Parse(args); err != nil {
+		return err
+	}
+
+	if *toFile == "" {
+		return fmt.Errorf("must provide an updated Thrift file")
+	}
+	// TODO: here we'll get into interesting stuff with git, but for now default to
+	// using originalFile as an argument.
```

Comment:
either provide `--to` and `--from` or `--repo` argument.

Reply:
How about `--base $hash` to specify the base to compare against? It could default to `HEAD^` (one commit before now) or alternatively `HEAD` (most recent commit). The latter could be useful to compare a dirty working tree against the most recent commit. If we call it `--base`, then the "destination" should probably be called `--head` (base and head), and that can default to "current working tree".

So like, locally:

```
$ git status
[clean]
$ vim foo.thrift
[modify]
$ thriftbreak
# Will compare dirty working tree to HEAD which was the last commit.
foo.thrift:2:great sadness
```

And in CI:

```
$ git status
[clean]

$ base=$(git rev-parse HEAD)

# apply the patch
$ arc patch D1234   # or
$ gh pr checkout 123

$ head=$(git rev-parse HEAD)

$ thriftbreak --base "$base" --head "$head"
foo.thrift:2:great sadness
```